godal.go

```diff
@@ -3733,6 +3808,65 @@ func BuildVRT(dstVRTName string, sourceDatasets []string, switches []string, opt
 	return &Dataset{majorObject{C.GDALMajorObjectH(hndl)}}, nil
 }
 
+// GridCreate, creates a grid from scattered data, given provided gridding parameters as a string (pszAlgorithm)
+// and the arguments required for `godalGridCreate()` (binding for GDALGridCreate)
+//
+// NOTE: For valid gridding algorithm strings see: https://gdal.org/programs/gdal_grid.html#interpolation-algorithms
+func GridCreate(pszAlgorithm string,
+	numCoords uint32,
+	xCoords []float64,
+	yCoords []float64,
+	zCoords []float64,
+	dfXMin float64,
+	dfXMax float64,
+	dfYMin float64,
+	dfYMax float64,
+	nXSize uint32,
+	nYSize uint32,
+	buffer interface{},
+	opts ...GridCreateOption,
+) ([]byte, error) {
```

Comment:
The `[]byte` return value is not desired here. i.e. GridCreate should return a plain error, as the resulting grid is present in `buffer`